### ISSUE TITLE
Web Assembly doesn't work after following documentation

### ISSUE THREAD

**Issue by davidgiga1993** [AvaloniaUI/Avalonia#16989]
### Describe the bug

When following the steps described in the [documentation](https://docs.avaloniaui.net/docs/guides/platforms/how-to-use-web-assembly) and running the web server using `dotnet run` the web app can't load due to `404 http://127.0.0.1:62860/_framework/avalonia.js`

### To Reproduce

Follow the guide, effectively:

```
dotnet workload install wasm-tools
dotnet new install avalonia.templates
mkdir BrowserTest
cd BrowserTest
dotnet new avalonia.xplat
cd BrowserTest.Browser
dotnet run
```

### Expected behavior

The sample app should work

### Avalonia version

11.1.0

### OS

WebAssembly

### Additional context

The template used `<TargetFramework>net8.0-browser</TargetFramework>`, but I also tested with `net9.0-browser`.

I've also noticed that the path described in the documentation under `Testing AppBundle locally` doesn't seem to be correct (or at least for me this doesn't match up at all, maybe it's related to the issue above?)

**Comment by maxkatz6** [comment]
Make sure you have up-to-date .NET SDK installed. I.e. latest patch releases. 
There were reports of similar problems on older .NET 8 builds.

**Comment by davidgiga1993** [comment]
Yep I actually just installed .net sdk 9 (rc1) when going through the document, so should be latest

**Comment by houstonhaynes** [comment]
FYI - I'm running into a *marginally* related issue with `dotnet serve` per the [instructions](https://docs.avaloniaui.net/docs/guides/platforms/how-to-use-web-assembly). This is confirmed when running dotnet-serve on either Windows or Linux - I get a "nice clean" console in the browser with no issues but the screen is simply blank after loading (the index.html pre-load/splash page loads as expected). 

Running locally from JetBrains Rider or `dotnet run` from the command line is fine, BTW. It's only when serving from the AppBundle 

```pwsh
dotnet publish -p:CompressionEnabled=false 
dotnet serve -d:./bin/Release/net8.0-browser/browser-wasm/AppBundle --port 5000 --address any 
```

Avalonia 11.1.0
dotnet sdk 8.0.4

![Screenshot 2024-09-16 133856](https://github.com/user-attachments/assets/0aea074e-71a2-41c0-940f-62593efc7d7e)

**Comment by hypeartist** [comment]
Same here. Latest.NET (RC1), Avalonia 11.1.0.
PS: After changing project target to net9-browser error changed from not finding avalina.js into:
![image](https://github.com/user-attachments/assets/7be4ddc0-2116-41ff-8f7f-cf3f6faeeb35)

**Comment by maxkatz6** [comment]
@davidgiga1993 you are running "net8.0" target on .NET 9 SDK. .NET Runtime team sometimes breaks compatibility between SDK versions, especially with WASM workloads.
If you want to run "net8.0", which is in stable, but using .NET 9 SDK, then you need to install **another workload** ` dotnet workload install wasm-tools-net8`. As confusing it is.
We probably should add a general advice in our documentation to keep these versions in sync.

I just tested, and `net8.0-browser` target works well with .NET 9 SDK with `wasm-tools-net8.` 

As for another issue when you switch to `net9.0-browser`, somebody of you should report it to https://github.com/dotnet/runtime.

**Comment by maxkatz6** [comment]
Just added wasm-tools-net8 note to our documentation https://github.com/AvaloniaUI/avalonia-docs/pull/548

**Comment by maxkatz6** [comment]
Also, thanks for noticing that `dotnet publish` output path was changed. I updated documentation accordingly.

**Comment by maxkatz6** [comment]
.NET 9 issue is most likely caused by .NET runtime migrating to a newer Emnscripten version: https://github.com/dotnet/runtime/pull/100334
We are blocked on SkiaSharp release, which should happen before .NET 9 stable release.

**Comment by gpatrut-matica** [comment]
I'm actually running in the same issue. I started with a clean slate. Ran all the commands in the documentation and when I went and tried running the browser project I got the missing "_framework/avalonia.js" issue. I did not change anything in the project. 

.NET SDK 8.0.400
Avalonia 11.1.0
Both projects have `<TargetFramework>net8.0</TargetFramework>` and `<TargetFramework>net8.0-browser</TargetFramework>`

I tried running it with .NET 9 and the `wasm-tools-net8` workload, no luck, altought I could not even build the project as I got a gigantic dump of text in the output console (none with useful information to be honest).

**Comment by maxkatz6** [comment]
@gpatrut-matica I can't help you without project itself, and the best I can advice:
1. Recreate Browser project from the template (it was changed between 11.0 and 11.1).
2. Make sure you have latest .NET 8 SDK patches
3. Latest Avalonia itself (11.1.0 is OLD)

**Comment by ztl8702** [comment]
I can confirm that .NET 9 SDK targeting `net8.0-browser` with `wasm-tools-net8` workload works for me. I am using the `Avalonia.XPlat` template from `Avalonia.Templates::11.1.0` and manually updated Avalonia version to 11.1.3.

I also noticed:
  * when targeting `net9.0-browser` and doing a `dotnet publish`, `avalonia.js` exists at the root of `wwwroot` instead of at `wwwroot/_framework`. 
    
    I assume this could be fixed by overriding [`FrameworkAssetPathResolver`](https://github.com/AvaloniaUI/Avalonia/blob/aabd038e88951a992124ddfcb2be06ee6daaf4a5/src/Browser/Avalonia.Browser/BrowserAppBuilder.cs#L35) but I didn't go further because I also noticed some SkiaSharp exceptions. Will wait for https://github.com/AvaloniaUI/Avalonia/issues/16989#issuecomment-2372785071
